WordPress/src/main/java/org/wordpress/android/ui/media/services/MediaUploadService.java

```diff
@@ -274,34 +231,38 @@ private void cancelUpload() {
     }
 
     private void dispatchUploadAction(@NonNull final MediaModel media) {
-        AppLog.i(T.MEDIA, "Dispatching upload action for media with local id: " + media.getId() +
+        AppLog.i(AppLog.T.MEDIA, "Dispatching upload action for media with local id: " + media.getId() +
                 " and path: " + media.getFilePath());
         media.setUploadState(UploadState.UPLOADING.name());
         mDispatcher.dispatch(MediaActionBuilder.newUpdateMediaAction(media));
 
         MediaPayload payload = new MediaPayload(mSite, media);
         mDispatcher.dispatch(MediaActionBuilder.newUploadMediaAction(payload));
-
-        if (mListener != null) {
-            mListener.onUploadBegin(mCurrentUpload);
-        }
     }
 
     private void dispatchCancelAction(@NonNull final MediaModel media) {
-        AppLog.i(T.MEDIA, "Dispatching cancel upload action for media with local id: " + media.getId() +
+        AppLog.i(AppLog.T.MEDIA, "Dispatching cancel upload action for media with local id: " + media.getId() +
                 " and path: " + media.getFilePath());
         MediaPayload payload = new MediaPayload(mSite, mCurrentUpload);
         mDispatcher.dispatch(MediaActionBuilder.newCancelMediaUploadAction(payload));
     }
 
+    private void completed(){
+        AppLog.i(AppLog.T.MEDIA, "Media Upload Service > completed");
+        if (mQueue == null || mQueue.size() == 0) {
```

Comment:
You're right it is being called in those places as of now, however I prefer to have one exit point where we do any clean up and call `stopSelf()`, rather than having several places where stopSelf could be called without check (or duplicating checks) - also important to note is this is being enhanced in another PR to handle multiple uploads